internal/provider/kubernetes/controller.go

```diff
@@ -940,7 +940,13 @@ func (r *gatewayAPIReconciler) processGateways(ctx context.Context, managedGC *g
 
 		if err := r.processGatewayParamsRef(ctx, &gtw, resourceMap, resourceTree); err != nil {
 			r.log.Error(err, "failed to process infrastructure.parametersRef for gateway", "namespace", gtw.Namespace, "name", gtw.Name)
+			status.SetGatewayAccepted(&gtw, false, gwapiv1.GatewayReasonInvalidParameters, err.Error())
+			r.updateGatewayStatus(&gtw)
```

Comment:
`r.updateGatewayStatus` is used to  trigger a dummy Gateway Status update event to update the gateway IP address and can't be used for update Accepted status.

I suggest to add the Accepted status to the Gateway resource and pass it to the Gateway API translator to handle it.